android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/views/UserLocationView.java

```diff
@@ -574,7 +574,7 @@ private void setLocation(Location location) {
             mShowDirection = true;
             mGpsMarkerDirection = 0;
             if (location.hasBearing()) {
-                mMapView.setBearing(mUserLocation.getBearing());
+                mMapView.setBearing(location.getBearing());
```

Comment:
@bleege I'm not very familiar with that class, but moving the assignment earlier will change the behavior of other blocks of code like https://github.com/mapbox/mapbox-gl-native/blob/a0f2553ab4af7e9723e1e0723bb070f665321aeb/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/views/UserLocationView.java#L581-L589 that rely on a previous value of `mUserLocation`.